cyipopt/scipy_interface.py

```diff
@@ -486,13 +486,11 @@ def minimize_ipopt(fun,
     (fun, x0, args, kwargs, method, jac, hess, hessp,
      bounds, constraints, tol, callback, options) = res
 
-    _x0 = np.atleast_1d(x0)
```

Comment:
Conversion to 1d is now done in `_minimize_ipopt_iv`.

Previously, the variable was renamed to `_x0` so that at the end, if `x0` had been a scalar, the `x` attribute of the result object would be a scalar.

In SciPy, I am very careful about shapes, and I have corrected many shape bugs related to this sort of thing in SciPy code. However, I'm not sure if this is needed here. 

The documentation (which now mirrors that of [`scipy.optimize.minimize`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.optimize.minimize.html)) states that `x0` should be an array-like of shape `(n,)`, where `n` is the number of independent variables. This function does not handle arrays of arbitrary dimensionality, and there is a separate `minimize_scalar` function for functions of a single variable. Also, the `x` attribute of the object returned by `scipy.optimize.minimize` returns 1d array even when x0 is a scalar:

```python3
from scipy.optimize import minimize

def f(x):
    return x**2

minimize(f, 1).x  # [-7.450e-09]
```

All these things suggest to me that the output should always be a 1d array (even if the function is flexible about accepting scalar input rather than raising an error)and working with functions that return scalars).

LMK if you'd prefer for this to be left alone.

Reply:
If our function here supported scalar and array objective functions simultaneously, I think we'd need to deprecate the scalar functionality with at least a release.

Reply:
This would not change what types of objective functions are supported. It makes the function more consistent with the documentation, which states:

![image](https://github.com/mechmotum/cyipopt/assets/6570539/32faf8af-4094-41dc-b630-468bf859c235)

In SciPy, the `x` attribute of the result object is _always_ at least 1D. However, currently in `minimize_ipopt`, the `x` attribute of the result object may be a scalar. The proposed change would resolve this inconsistency and make `minimize_ipopt` behave as advertised - like `scipy.optimize.minimize`.

Reply:
Ok. I consider this backwards incompatible, regardless of what the docs say it should do, but we can make the change with no deprecation.